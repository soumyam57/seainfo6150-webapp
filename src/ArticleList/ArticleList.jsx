import React from "react";
import PropTypes from "prop-types";
import styles from "./ArticleList.module.css"
import ArticleListItem from "../ArticleListItem/ArticleListItem.jsx";

const ArticleList = (props) => {
  let displayContent;

  if (props.articles.length) {
    displayContent = (   
        <ul className={styles.container}>
          {props.articles.map(article => (
            <li key={article.slug}>
              <ArticleListItem article={article} />
            </li>
          ))}
        </ul>    
        
    );
  } else {
    displayContent = <div>You have no data!</div>;
  }

  return (
    <div>
      {displayContent}
    </div>
  );
};

ArticleList.propTypes = {
    articles: PropTypes.array.isRequired
}
export default ArticleList;