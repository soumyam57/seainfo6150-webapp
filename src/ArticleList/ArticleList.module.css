.container {
  background: #999;
  border: 1px dashed #666;
  border-radius: 10px;
  box-shadow: 0px 0px 10px 5px rgba(204, 0, 0, 0.5);
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 20px;
  font-family: Arial, Helvetica, sans-serif;
  list-style-type: none;
  padding: 15px;
}

@supports (display: grid) {
  .container {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
  }

  @media screen and (min-width: 300px) {
    .container {
      grid-template-columns: repeat(1,1fr);
      background: lightslategray;
      padding: 40px;
      gap: 40px;
    }
  }

  @media screen and (min-width: 550px) {
    .container {
      grid-template-columns: repeat(2,1fr);
      background: lightsalmon;
      font-family: Verdana;
      font-style: oblique;
      padding: 25px;
      gap: 30px;
    }
  }

  @media (min-width: 900px) {
    .container {
      grid-template-columns: repeat(3, 1fr);
      background: #999;
      font-style: normal;
      padding: 15px;
      gap: 20px;
    }
  }
}
