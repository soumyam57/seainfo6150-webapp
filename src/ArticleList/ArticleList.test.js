import React from 'react';
import renderer from 'react-test-renderer';
import ArticleList from './ArticleList';

const articleArray = [
    {
        key: "181122102925-statue-of-liberty-crown"[
            {
                author: "Kate Farley",
                displayDate: "November 22nd 2018, 7:12 am",
                shortText: "The Statue of Liberty's original torch is getting a new home",
                title: "The Statue of Liberty's torch heads to new museum"
            }
        ]
    }
    ,
    {
        key: "170321135754-airport-security-line"[
        {
            author: "Cathryn Lindsey",
            displayDate: "November 21st 2018, 2:57 pm",
            shortText: "Getting excited to eat your mother's roast turkey, your uncle's stuffing and your sister's famous green bean casserole?",
            title: "You can take a pie on a plane"
        }
    ]
}]


describe("Articlelist tests", () => {
    it("renders correctly when passed an array", () => {
    const { container } = render(<ArticleList  articles={Object.values(articleArray)}  key={articleArray.key}/>);
      expect(container).toMatchSnapshot();
    });
});
