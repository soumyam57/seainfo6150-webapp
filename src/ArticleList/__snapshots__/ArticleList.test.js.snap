// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articlelist tests renders correctly when passed an array 1`] = `
<div>
  <div>
    <ul
      class="container"
    >
      <li
        class="container"
      >
        <article
          class="article"
        >
          <div
            class="wrapper"
          >
            <h2
              class="title"
            />
          </div>
          <button
            class="button"
          >
            Show more
          </button>
        </article>
      </li>
      <li
        class="container"
      >
        <article
          class="article"
        >
          <div
            class="wrapper"
          >
            <h2
              class="title"
            />
          </div>
          <button
            class="button"
          >
            Show more
          </button>
        </article>
      </li>
    </ul>
  </div>
</div>
`;
